actors/builtin/miner/miner_actor.go

```diff
@@ -466,9 +482,10 @@ func (a Actor) DeclareTemporaryFaults(rt Runtime, params DeclareTemporaryFaultsP
 	builtin.RequireSuccess(rt, code, "failed to burn fee")
 
 	// Request deferred Cron invocation to update temporary fault state.
+	// TODO: cant we just lazily clean this up?
```

Comment:
I guess you could do that when computing the proving set, that is: on submitting the fallback post, you need to subtract faults from the provingSet. You could check whether thee fault is still active then... Assuming there is no deeper incidence on collateral and such cc @zixuanzh